pkg/proxy/handlers/spacelister.go

```diff
@@ -81,68 +149,98 @@ func (s *SpaceLister) ListUserWorkspaces(ctx echo.Context) ([]toolchainv1alpha1.
 	return s.workspacesFromSpaceBindings(signup.Name, spaceBindings), nil
 }
 
-func (s *SpaceLister) listSpaceBindingsForUser(ctx echo.Context, murName string) ([]toolchainv1alpha1.SpaceBinding, error) {
+func (s *SpaceLister) listSpaceBindingsForUser(murName string) ([]toolchainv1alpha1.SpaceBinding, error) {
+	murSelector, err := labels.NewRequirement(toolchainv1alpha1.SpaceBindingMasterUserRecordLabelKey, selection.Equals, []string{murName})
+	if err != nil {
+		return nil, err
+	}
+	requirements := []labels.Requirement{*murSelector}
+	return s.GetInformerServiceFunc().ListSpaceBindings(requirements...)
+}
 
+func (s *SpaceLister) listSpaceBindingForUserAndSpace(ctx echo.Context, murName string) (*toolchainv1alpha1.SpaceBinding, error) {
 	workspaceName := ctx.Param("workspace")
-	doGetWorkspace := len(workspaceName) > 0
-
 	murSelector, err := labels.NewRequirement(toolchainv1alpha1.SpaceBindingMasterUserRecordLabelKey, selection.Equals, []string{murName})
 	if err != nil {
 		return nil, err
 	}
+	// specific workspace requested so add label requirement to match the space
+	spaceSelector, err := labels.NewRequirement(toolchainv1alpha1.SpaceBindingSpaceLabelKey, selection.Equals, []string{workspaceName})
+	if err != nil {
+		return nil, err
+	}
+	requirements := []labels.Requirement{*murSelector, *spaceSelector}
 
-	requirements := []labels.Requirement{*murSelector}
+	spaceBindings, err := s.GetInformerServiceFunc().ListSpaceBindings(requirements...)
+	if err != nil {
+		return nil, err
+	}
 
-	if doGetWorkspace {
-		// specific workspace requested so add label requirement to match the space
-		spaceSelector, err := labels.NewRequirement(toolchainv1alpha1.SpaceBindingSpaceLabelKey, selection.Equals, []string{workspaceName})
-		if err != nil {
-			return nil, err
-		}
-		requirements = append(requirements, *spaceSelector)
+	//  let's only log the issue and consider this as not found
+	if len(spaceBindings) != 1 {
```

Comment:
we could change that so when 
`len(spaceBindings) == 0` then we would return not found
and if `len(spaceBindings) > 1` then we would return "internal error"

Reply:
Good point! Done with tests: https://github.com/codeready-toolchain/registration-service/pull/348/commits/01759250a7329b5df84baeca99ea984bd28029ec